lib/notifier.js

```diff
@@ -280,6 +280,42 @@ function addListener(server, user, regex, event, callback) {
     return listener.add(_config.filterRegex, _event, _callback);
 }
 
+function addStartupListener(config, callback) {
+    if (!config.serverUrl || !config.adminUser || !config.filterRegex) {
+        throw new Error(`Invalid 'config' parameter ${JSON.stringify(config)},'serverUrl', 'adminUser' and 'filterRegex' are required parameters`);
+    }
+    // let config = {
+    //     serverUrl: server,
+    //     adminUser:  user,
+    //     filterRegex: regex,
+    //     sslConfiguration: sslConfiguration
+    // };
+
+    if (!listener) {
+        listener = new Listener(this, config);
+    }
+
+    const realmAvailableCallback = (realmPath, realmId) => {
+        const absolutePath = require('path').join(require('process').cwd(), 'realm-object-server/listener/realms', realmPath, `${realmId}.realm`);
+        if (require('fs').existsSync(absolutePath)) {
+            new Promise(async (resolve) => {
+                var realm = new Realm({
+                    path: absolutePath,
+                    sync: true
+                });        
+                await callback(realmPath, realm); // callback could be an async or sync function.  
+                realm.close(); // close the Realm to avoid realm::MismatchedConfigException realm already opened with different sync configurations.
```

Comment:
If the user's callback is async then this is far too late to close the realm, as the GN will have already tried to open the realm with a different configuration before this is reached.